### PR TITLE
Update nginx and ngx_mruby

### PR DESCRIPTION
**Pull request by workmad3**
**Trello:** https://trello.com/c/8TpvxpR2/718-update-to-nginx-1-11-3

Updates nginx and ngx_mruby to their latest versions. 

The build process for ngx_mruby changed very slightly in the latest release, requiring a small change to our customised build_config file.

### CODE REVIEW

File: mruby_build_config.rb [hunk 1]
```diff
@@ -13,6 +13,10 @@
 
   conf.gembox 'full-core'
 
+  conf.cc do |cc|
+    cc.flags << ENV['NGX_MRUBY_CFLAGS'] if ENV['NGX_MRUBY_CFLAGS']
+  end
```
Comment (harimohanraj89) [inline mruby_build_config.rb]: What does this ENV var do?

**Comment by harimohanraj89** [comment]
I have only one curiosity question. This looks good! 👍

File: mruby_build_config.rb (hunk 1 above)
Comment (workmad3) [inline mruby_build_config.rb]: It passes on some compiler flags... we're not currently using it (unless it's used inside the mruby build script) but it's been added to `build_config.rb` upstream, so I figured we should copy it across to our customised version :)

**Comment by workmad3** [comment]
Unfortunately, if we put it on staging (by editing the buildpack url), we end up with it in production with the first promote anyway... I've tested it in dev to make sure the config is valid and nothing has broken, so I'm tempted to merge and let it sit in staging until we have a PR to promote (so I won't actively push to production just for the upgrade).

**Comment by harimohanraj89** [comment]
Ah I see, good point. That sounds like a good plan to me. We _do_ have an open PR in Switchboard, so we should be mindful of how soon we get that merged in :)

**Pull request merged**